zeebe/engine/src/test/java/io/camunda/zeebe/engine/processing/bpmn/activity/listeners/execution/ExecutionListenerMultiInstanceActivitiesTest.java

```diff
@@ -0,0 +1,246 @@
+import io.camunda.zeebe.test.util.record.RecordingExporterTestWatcher;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+import java.util.function.BiConsumer;
+import java.util.function.Function;
+import org.assertj.core.groups.Tuple;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.junit.runners.Parameterized.Parameter;
+import org.junit.runners.Parameterized.Parameters;
+
+@RunWith(Parameterized.class)
+public class ExecutionListenerMultiInstanceActivitiesTest {
+  @ClassRule public static final EngineRule ENGINE = EngineRule.singlePartition();
+  private static final Collection<Integer> INPUT_COLLECTION = List.of(1, 2, 3, 4);
+
+  @Rule
+  public final RecordingExporterTestWatcher recordingExporterTestWatcher =
+      new RecordingExporterTestWatcher();
+
+  @Parameter public MultiInstanceTestScenario scenario;
+
+  @Parameters(name = "{index}: {0}")
+  public static Collection<Object[]> parameters() {
+    return Arrays.asList(
+        new Object[][] {
+          {
+            MultiInstanceTestScenario.of(
+                "service_task",
+                BpmnElementType.SERVICE_TASK,
+                se -> se.serviceTask("service_task").zeebeJobType(SERVICE_TASK_TYPE),
+                (pik, index) -> completeJobByType(pik, SERVICE_TASK_TYPE, index))
+          },
+          {
+            MultiInstanceTestScenario.of(
+                "receive_task",
+                BpmnElementType.RECEIVE_TASK,
+                se ->
+                    se.receiveTask("receive_task")
+                        .message(mb -> mb.name("msg").zeebeCorrelationKey("=\"id-123\"")),
+                (pik, index) ->
+                    ENGINE.message().withName("msg").withCorrelationKey("id-123").publish())
+          },
+          {
+            MultiInstanceTestScenario.of(
+                "manual_task", BpmnElementType.MANUAL_TASK, se -> se.manualTask("manual_task"))
+          },
+          {
+            MultiInstanceTestScenario.of(
+                "user_task",
+                BpmnElementType.USER_TASK,
+                se -> se.userTask("user_task"),
+                (pik, index) -> completeJobByType(pik, "io.camunda.zeebe:userTask", index))
+          },
+          {
+            MultiInstanceTestScenario.of(
+                "sub_process",
+                BpmnElementType.SUB_PROCESS,
+                se ->
+                    se.subProcess(
+                        "sub_process",
+                        s -> s.embeddedSubProcess().startEvent().manualTask().endEvent()))
+          },
+          {
+            MultiInstanceTestScenario.of(
+                "call_activity",
+                BpmnElementType.CALL_ACTIVITY,
+                se -> se.callActivity("call_activity", s -> s.zeebeProcessId(SUB_PROCESS_ID)))
+          }
```

Comment:
:wrench: We could reduce the number of test cases.

Since the multi-instance behavior is the same for all elements, verifying the EL behavior with one element type (e.g., with service tasks) would be enough.

Reply:
Done ☑️, simplified this test class by removing parameterization, 
leaving checks for  the `service task` element type.